cpp/src/IceGrid/RegistryI.cpp

```diff
@@ -242,7 +242,7 @@ RegistryI::startImpl()
     }
 
     setupThreadPool(properties, "IceGrid.Registry.Client.ThreadPool", 1, 10);
-    setupThreadPool(properties, "IceGrid.Registry.Server.ThreadPool", 1, 10, true); // Serialize for admin callbacks
```

Comment:
TODO: Do we need to set MaxDispatches=1 on this OA?

It's not immediately clear to me what this is about.

Reply:
It's a though one to answer :slightly_smiling_face:.

It's about the following: https://doc.zeroc.com/ice/3.7/ice-services/icegrid/icegrid-and-the-administrative-facility#id-.IceGridandtheAdministrativeFacilityv3.7-CallbackswithoutGlacier2

In the example, the status of the services of an IceBox server can be observed by an IceGrid admin client. The IceGrid Admin GUI supports this, it shows the status of services from an IceBox server.

Here, it's important that calls performed by the IceBox server (https://github.com/zeroc-ice/ice/blob/146959f2079d689a209708c16d2f60c1f810bbbd/cpp/src/IceBox/ServiceManagerI.cpp#L819) to the IceGrid admin client service observer are received in the same order.

These calls are received on the IceGrid registry server endpoints and forwarded to the admin client. The serialize mode ensures that the calls are forwarded in the same order.

So `MaxDispatches=1` is indeed needed...

Reply:
> Here, it's important that calls performed by the IceBox server (...code...) to the IceGrid admin client service observer are received in the same order.

The concern is ServiceManagerI could call servicesStartedAsync and servicesStoppedAsync on the _same_ observer, and we want these non-awaited calls to reach the ultimate observer in the same order?